TechnicoBackend/Controllers/PropertyController.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using TechnicoBackend.Services;
+using TechnicoBackend.Models;
+using TechnicoBackend.DTOs;
+
+namespace TechnicoBackend.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class PropertyController : ControllerBase
+    {
+        private readonly PropertyService _propertyService;
+
+        public PropertyController(PropertyService propertyService)
+        {
+            _propertyService = propertyService;
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetProperties()
+        {
+            try
+            {
+                if (!Request.Cookies.TryGetValue("userId", out var userId))
+                {
+                    return Unauthorized("Ο χρήστης δεν είναι συνδεδεμένος.");
+                }
+
+                var properties = await _propertyService.GetPropertiesByUserIdAsync(Guid.Parse(userId));
+                return Ok(properties);
+            }
+            catch (Exception ex)
+            {
+                return BadRequest($"Σφάλμα κατά την ανάκτηση των properties: {ex.Message}");
+            }
+        }
+
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetProperty(Guid id)
+        {
+            try
+            {
+                var property = await _propertyService.GetPropertyByIdAsync(id);
+                return Ok(property);
+            }
+            catch (KeyNotFoundException ex)
+            {
+                return NotFound(ex.Message);
+            }
+            catch (Exception ex)
+            {
+                return BadRequest(ex.Message);
+            }
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> CreateProperty([FromBody] PropertyDTO propertyDto)
```

Comment:
Consider using a Service to implement any business logic/functionality. Consider using Controller to only receive the data and call the service.